src/types/timestamp.rs

```diff
@@ -75,7 +75,11 @@ impl Mantissa {
             Ordering::Equal
         } else if d1.coefficient.is_zero() && d2.coefficient.is_zero() {
             // Coefficient zeros' signs don't have to be compared for fractional seconds.
-            d1.exponent.cmp(&d2.exponent)
+            if d1.coefficient.sign() == Sign::Positive {
+                d2.exponent.cmp(&d1.exponent)
+            } else {
+                d1.exponent.cmp(&d2.exponent)
```

Comment:
Could you explain this change a bit? I'm unsure why the ordering logic would have changed.

Reply:
I updated this branch to return `Ordering::Equal`, since `Mantissa::decimals_compare` is only used for testing fractional seconds fields for ordering, and we won't care for the precision once we know that both coefficients are zero